pyflakes/checker.py

```diff
@@ -1220,8 +1222,8 @@ def addArgs(arglist):
         def runFunction():
 
             self.pushScope()
-            for name in args:
-                self.addBinding(node, Argument(name, node))
+
+            self.handleNode(node.args, node)
             if isinstance(node.body, list):
```

Comment:
Now you should also be able to replace this (and the above handleNode) with `self.handleChildren(node, omit='decorator_list')` .

the decorators are already processed at the beginning of the `FunctionDef` handler, as they are in the outer scope.

Reply:
It is a single node. Why to use `handleChildren`?

Reply:
No, use `self.handleChildren(node, omit='decorator_list')` and replace the `if` block on *this* line.  The body of the node can now be routed through `handleChildren` because of the other improvements you have made.